tests/cpp/test_circular_buffering.cpp

```diff
@@ -1266,11 +1266,16 @@ TEST_P(TmaCircularBufferingTest, Pointwise) {
   tv3->circularBuffer(
       number_of_stages, prefetch_distance, circular_buffer_type);
 
-  // Circular Buffer with set operation
-  // Load TV1 into shared memory
-  tv4->setMemoryType(MemoryType::Shared);
-  tv4->circularBuffer(
-      number_of_stages, prefetch_distance, circular_buffer_type);
+  // Circular Buffer with set operation.
+  // Note that in order to use warp specialization, all circilar buffers must be
```

Comment:
Why is this limitation?

Reply:
We're using `mbarrier::arrive-expect-tx` in the synchronization between load and compute warps, which requires using TMA load operations.

Basically, we're going from pipelined, mbarrier-based tma loads to a warp specialized kernel.

Mbarrier-based mechanism Reference: https://docs.nvidia.com/cuda/parallel-thread-execution/#completion-mechanisms-for-asynchronous-copy-operations

`cp.async.bulk` completion mechanisms: https://docs.nvidia.com/cuda/parallel-thread-execution/#data-movement-and-conversion-instructions-cp-async-bulk

Reply:
Ryan is correct. We are missing a mechanism to wait for the completion of set op, so we will not have RAW wait for it. We can add one of course, but because I don't think anyone will be interested in using set-op with warp specialization, I will just leave this as an error.

Reply:
Got it, thanks!